src/alr/alr-commands-build.adb

```diff
@@ -15,13 +17,24 @@ package body Alr.Commands.Build is
 
    procedure Do_Compile is
    begin
+      Requires_Full_Index;
```

Comment:
@mosteo I am not sure about the consequences of calling this procedure here.
I guess this will not fetch the index from the web, but what are the consequences in terms of performances?

Reply:
That loads all the local indexes. It does not reload unless explicitly forced, so it is safe to call at any point where we want the crates to be already loaded.

In the case of `alr build`, it is there so post-compile actions are known and run. Obviously we could make things more efficient by recursively loading dependencies only, or caching actions after retrieval... I'm actually unsure if any crate uses `post-compile` actions.

I guess as the index grows bigger, loading time may become noticeable and we may need to do something about it.